internal/uvm/computeagent.go

```diff
@@ -46,6 +50,44 @@ func (ca *computeAgent) AddNIC(ctx context.Context, req *computeagent.AddNICInte
 	return &computeagent.AddNICInternalResponse{}, nil
 }
 
+// ModifyNIC will modify a NIC from the computeagent services hosting UVM.
+func (ca *computeAgent) ModifyNIC(ctx context.Context, req *computeagent.ModifyNICInternalRequest) (*computeagent.ModifyNICInternalResponse, error) {
+	log.G(ctx).WithFields(logrus.Fields{
+		"nicID":        req.NicID,
+		"endpointName": req.EndpointName,
+	}).Info("ModifyNIC request")
+
+	if req.IovPolicySettings == nil {
```

Comment:
Check for empty endpoint name and NicID also here

Reply:
Do you want me to add similar checks to the other functions in this file as well?

Reply:
Yea any that rely on all the settings being passed to be a valid request

Reply:
Done